primitives/src/identity.rs

```diff
@@ -0,0 +1,58 @@
+use codec::{Decode, Encode};
+use rstd::prelude::Vec;
+
+use crate::{IdentityId, IdentityRole, Key, SigningKey};
+
+/// Identity information.
+#[allow(missing_docs)]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug)]
+pub struct Identity {
+    pub roles: Vec<IdentityRole>,
+    pub master_key: Key,
+    pub signing_keys: Vec<SigningKey>,
+    pub signing_identities: Vec<IdentityId>,
```

Comment:
I'm wondering why we need to store `signing_identities` separately to `signing_keys` as `SigningKey` already has a type that covers identities:
```
pub enum KeyType {
    External,
    Identity,
    Multisig,
    Relayer,
    Custom(u8),
}
```

Reply:
After a chat with @maxsam4 , I added that in order to mitigate the constraint 1-to-1 between keys and Identities. The issue is that any master key cannot be a signing key of other identity. 
That `signing_identities` will also need roles/permissions (same way we already have in `signing_keys`), but we kept it simple until the permission spec would be complete.